Mapbox/MapboxMapsLocation/LocationManager.swift

```diff
@@ -86,26 +86,20 @@ public class LocationManager: NSObject {
 
         guard newOptions != locationOptions else { return }
 
+        // Update the location options
+        self.locationOptions = newOptions
+        self.locationProvider.locationProviderOptions = newOptions
+
         if newOptions.showUserLocation != showUserLocation {
             self.showUserLocation = newOptions.showUserLocation
             toggleUserLocationUpdates(showUserLocation: self.showUserLocation)
-
-            if !newOptions.showUserLocation {
-                // If we should not show user location, then we should
-                // not try and change source or style below
-                return
-            }
```

Comment:
This wasn't providing any sort of optimization.

Reply:
what happens in the case if someone decides to stop showing user location, and for some reason they have change another option like location puck or something below?

Reply:
any reason why those changes should not take effect? This `return` feels like premature optimization.